md/1157.md

```diff
@@ -0,0 +1,45 @@
+ข้อนี้เป็นโจทย์ Dynamic Programming แบบ Sliding Window
+
+# Naive Dynamic Programming Solution 
+แนวคิดพื้นฐานของข้อนี้คือใช้ Dynamic Programming ในการคำนวณ $min_cost[i]$ ซึ่งนิยามเป็นค่าใช้จ่ายที่ต่ำที่สุดที่จะทำให้สามารถต่อจากแปลงที่ $1$ ไปยังแปลงที่ $i$ (รวมถึงสร้างแปลงที่ $i$)
+
+การคำนวน $min_cost[i]$ ต้องพิจารณาเพียงแปลง $i-k$ ถึงแปลง $i-1$ และเลือกอันที่มี $min_cost$ ต่ำสุด
+
+แนวคิดนี้สามารถเขียนเป็นโค้ด:
+```cpp
+min_cost[1] = P[1];
+for (int i = 2; i <= N; i++) {
+  min_cost[i] = 1000000001; // inf
+  for(int j = i - k; j < i; j++) {
+    min_cost[i] = min(min_cost[i], min_cost[j] + P[i]); 
+  }
+}
+```
+
+แต่วิธีนี้มี Time Complexity เป็น $\mathcal{O}(Nk)$ ซึ่งมากเกินไปสำหรับ $N=500000, k=2000$
+
+# Sliding Window 
+สังเกตว่าสามารถใช้ Sliding Window เก็บค่าต่ำสุดใน window ขนาด $k$ แทนการ iterate ทุกรอบเพื่อหาค่าต่ำสุด (อ่านเพิ่มได้ที่ https://usaco.guide/gold/sliding-window?lang=cpp - Deque)
+
+Code
```

Comment:
Explain how to code works. For general guidelines on how to explain please refer to [o62_may09_judtaew/solution](https://programming.in.th/tasks/o62_may09_judtaew/solution)

Reply:
Added Explanation (though it is short as the code itself is short)